sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsExtensionConfigProvider.cs

```diff
@@ -205,6 +210,35 @@ public async Task<HttpResponseMessage> ConvertAsync(HttpRequestMessage input, Ca
             return (T)blob.BlobClient;
         }
 
+        private static ParameterBindingData ConvertToParameterBindingData(BlobAttribute blobAttribute)
+        {
+            var blobPath = BlobPath.ParseAndValidate(blobAttribute.BlobPath);
+
+            dynamic blobDetails = new ExpandoObject();
+            blobDetails.Connection = blobAttribute.Connection ?? Constants.DefaultAzureStorageConnectionName;
+            blobDetails.BlobName = blobPath.BlobName;
+            blobDetails.ContainerName = blobPath.ContainerName;
+
+            return CreateParameterBindingData(blobDetails);
+        }
+
+        private static ParameterBindingData ConvertToParameterBindingData(BlobBaseClient input, BlobTriggerAttribute blobAttribute)
+        {
+            dynamic blobDetails = new ExpandoObject();
```

Comment:
Because the value we serialize is part of a public contract that we can't break, I'm thinking we should formalize it by using a concrete type rather than using ExpandoObject. The latter may make people think they can remove things w/o effect.

Reply:
That's a fair point, kind of similar to what we're doing on the dotnet worker side (minus the abstract class) 👍